esmvalcore/preprocessor/__init__.py

```diff
@@ -660,7 +667,11 @@ def _apply_multimodel(products, step, debug):
         step,
         "\n".join(str(p) for p in products - exclude),
     )
-    result = preprocess(products - exclude, step, **settings)
+    result: list[PreprocessorFile] = preprocess(  # type: ignore
+        products - exclude,  # type: ignore
```

Comment:
```suggestion
        products - exclude,
```
this ignore can probably go if you use `Sequence` instead of `list` in the `preprocess` function signature

Reply:
Removing this gives

```
esmvalcore/preprocessor/__init__.py:676: error: Argument 1 to "preprocess" has incompatible type "set[PreprocessorFile]"; expected "Sequence[PreprocessorFile | Any | str | Path]"  [arg-type]
```